src/PlusDataCollection/StealthLink/vtkPlusStealthLinkTracker.cxx

```diff
@@ -917,10 +987,36 @@ PlusStatus vtkPlusStealthLinkTracker::InternalUpdate()
         this->ToolTimeStampedUpdateWithoutFiltering(toolIterator->second->GetSourceId(), instrumentToTrackerTransform, TOOL_OUT_OF_VIEW, unfilteredTime, timeSystemSec);
       }
     }
+    // if the wanted Transform is any tool to Tracker that is not considered the current instrument
     else
     {
       vtkSmartPointer<vtkMatrix4x4> transformMatrixForNotTrackedTool = vtkSmartPointer<vtkMatrix4x4>::New();
-      this->ToolTimeStampedUpdateWithoutFiltering(toolIterator->second->GetSourceId(), transformMatrixForNotTrackedTool, TOOL_OUT_OF_VIEW, unfilteredTime, timeSystemSec);
+      // if we are tracking the current instrument at all times
+      if (STRCASECMP(toolIterator->second->GetCustomProperty("AlwaysTrack").c_str(), "TRUE") == 0)
+      {
+        MNavStealthLink::Instrument instrument;
+        if (!this->InternalShared->GetInstrumentData(instrument, toolIterator->second->GetPortName()))    //thread safe
+        {
+          return PLUS_FAIL;
+        }
+
+        vtkPlusStealthLinkTracker::vtkInternal::GetInstrumentInformation(instrument, instrumentOutOfView, instrumentToTrackerTransform);
+        if (instrumentOutOfView == false)

```

Comment:
I know that instrumentOutOfView has been in the code, but It would make the code easier to read if we didn't use negative terms. It would be more clear to write "instrumentInTrackerView" than "instrumentOutOfView == false". We don't have to change it now, but we should consider. Or, instead of a binary flag, we could use TOOL_OK, TOOL_OUT_OF_VIEW enum.